src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Represents the Storage Object containing logic to writing and storing task list to memory.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Initializes Storage object to be used for saving and loading task list throughout Duke.
+     *
+     * @param filePath specified file for Storage object to read from and write to.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    /**
+     * Handles logic where ./data or duke.txt does not exists by creating them.
+     *
+     * @param savedTasks File Object for loading and saving tasks.
+     * @throws IOException if File object cannot create new file.
+     */
+    private void handleFileOrDirectoryDoesNotExist(File savedTasks) throws IOException {
+        if (savedTasks.getParentFile().mkdirs()) {
+            System.out.println("./data directory created");
+        }
+        if (savedTasks.createNewFile()) {
+            System.out.println("duke.txt created in ./data");
+        }
+    }
+
+    /**
+     * Saves tasks list to memory.
+     *
+     * @param taskItems tasks list to be saved to memory.
+     * @throws DukeException if task cannot be written to memory if file created in path.
+     */
+    public void saveTaskToMemory(List<Task> taskItems) throws DukeException {
+        File savedTasks = new File(filePath);
+        try {
+            if (!savedTasks.exists()) {
+                handleFileOrDirectoryDoesNotExist(savedTasks);
```

Comment:
Very good piece of code, good use of abstraction. But perhaps the naming of certain methods could be a bit more short?. Hahha i'm really scraping the bell here because I can't find anything wrong